### PR TITLE
Make batch span processor flush span queue during shutdown

### PR DESCRIPTION
**Pull request by andremarianiello**
Fixes https://github.com/open-telemetry/opentelemetry-go/issues/2563

### CODE REVIEW

File: sdk/trace/batch_span_processor.go [hunk 1]
```diff
@@ -145,6 +145,10 @@ func (bsp *batchSpanProcessor) OnEnd(s ReadOnlySpan) {
 func (bsp *batchSpanProcessor) Shutdown(ctx context.Context) error {
 	var err error
 	bsp.stopOnce.Do(func() {
+		err = bsp.ForceFlush(ctx)
+		if err != nil {
+			return
```
Comment (Aneurysm9) [inline sdk/trace/batch_span_processor.go:150]:
```suggestion
			return err
```

**Review by Aneurysm9 — COMMENTED** [review]
Please add a test to ensure that `ForceFlush()` is invoked by `Shutdown()` and a `CHANGELOG.md` entry describing the change.

File: sdk/trace/batch_span_processor.go (hunk 1 above)
Comment (andremarianiello) [reply sdk/trace/batch_span_processor.go:150]: That `return` returns from the anonymous `Do` parameter, not from `Shutdown`.

Comment (Aneurysm9) [reply sdk/trace/batch_span_processor.go:150]: You're right, my mistake.

**Comment by andremarianiello** [comment]
> Please add a test to ensure that `ForceFlush()` is invoked by `Shutdown()` and a `CHANGELOG.md` entry describing the change.

Working on the test

**Comment by andremarianiello** [comment]
@Aneurysm9 Looking at the code some more, it appears that `drainQueue` happens during `Shutdown`, which flushes spans in the queue, so a call to `ForceFlush` shouldn't be needed. I'm still investigating.

**Comment by MrAlias** [comment]
Is this still needed, or can it be closed?

**Comment by Aneurysm9** [comment]
It should not be needed.  As long as the queue is blocked and drained, that includes the effect of `ForceFlush()` as the spec anticipates.  I'll close this and it can be re-opened if needed later.

**Pull request closed without merging**